test/units/contrib/inventory/test_ec2_inventory.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+
+import pytest
+
+# contrib's dirstruct doesn't contain __init__.py files
+# import ec2 inventory manually
+root_dir = os.path.dirname(
+    os.path.dirname(
+        os.path.dirname(
+            os.path.dirname(os.path.dirname(os.path.abspath(__file__))))))
+inventory_dir = os.path.join(root_dir, 'contrib', 'inventory')
+inventory_dir_abspath = os.path.abspath(inventory_dir)
+sys.path.append(inventory_dir_abspath)
```

Comment:
It's not safe to modify sys.path outside of unit test functions, as these changes will effect other unrelated tests.

Reply:
You may be able to modify sys.path, perform an import, and then restore sys.path. That will probably prevent it from causing issues for other unit tests, but I haven't verified that.

Reply:
yeah, now I use `teardown_module` in `pytest` to restore sys.path.

```python
def teardown_module(module):
    # cleanup so that path is not polluted with other scripts
    if inventory_dir_abspath in sys.path:
        sys.path.remove(inventory_dir_abspath)
```

Reply:
I found a better way to handle the import, using a pytest fixture.

I have a PR (https://github.com/ansible/ansible/pull/46732) to fix the vmware_inventory unit tests that provides a good example:

https://github.com/ansible/ansible/pull/46732/files#diff-7a31f0c775c6683a8212d1f175db50e9R28

The `BaseException` handling was only needed because the inventory script uses `sys.exit` for import failures.

Reply:
cool! fixed as your way